pkg/common/common.go

```diff
@@ -172,19 +172,19 @@ func MarshallNamespace(gwKey client.ObjectKey, domain string) string {
 }
 
 func UnMarshallObjectKey(keyStr string) (client.ObjectKey, error) {
-	keySplit := strings.Split(keyStr, string(NamespaceSeparator))
-	if len(keySplit) < 2 {
-		return client.ObjectKey{}, fmt.Errorf("failed to split on %s: '%s'", string(NamespaceSeparator), keyStr)
+	namespaceEndIndex := strings.IndexByte(keyStr, NamespaceSeparator)
+	if namespaceEndIndex < 0 {
+		return client.ObjectKey{}, fmt.Errorf(fmt.Sprintf("failed to split on %s: '%s'", string(NamespaceSeparator), keyStr))
 	}
 
-	return client.ObjectKey{Namespace: keySplit[0], Name: keySplit[1]}, nil
+	return client.ObjectKey{Namespace: keyStr[:namespaceEndIndex], Name: keyStr[namespaceEndIndex+1:]}, nil
```

Comment:
I would like to see tests with:

* keyStr -> `#` which makes `namespaceEndIndex` to be `0` and then what happens with `keyStr[namespaceEndIndex+1:]` ?

* keyStr -> `something#` which makes `namespaceEndIndex` to be the last char and then what happens with `keyStr[namespaceEndIndex+1:]` ?

Reply:
Could you please check out my PR with the test-cases for UnMarshallObjectKey? 

I have the following test-case there, which includes the two cases you mentioned:
```
{
    name: "when '#' separator used instead of '/' then return an error",
        input: "default#object1",
        expectedOutput: client.ObjectKey{},
        expectedError:  fmt.Errorf("failed to split on %s: 'default#object1'", string(NamespaceSeparator)),
},
```

When there is no `NamespaseSeparator', which is defined by constant, then it immediately returns an error in all cases. Here, a non-default separator '#' is used, so it won't get to the last return statement with slicing.

But I'll porbably rename it, by adding word 'default separator', because separator may possibly be changed? Anyway, it will make tests less dependant on implementation, I think.